### ISSUE TITLE
Support for literal assignments in body

### ISSUE THREAD

**Issue by Laschoking** [knowsys/nemo#353]
Consider the following  code snippet, which currently does not work:
```
literalInBody(?a):-   
    ?a = "test".
```
Throws following error:

```rust
 ERROR nmo] error: Parse error on line 26, column 1: Expected a statement (i.e., either a fact or a rule)  
    at "literalInBody(?a):-"
    ----^
    Context:
      Parse error on line 26, column 1: Expected a rule
      at "literalInBody(?a):-"
      ----^
      Context:
        Parse error on line 26, column 1: The universal variable "a" does not occur in a positive body literal.  
      at "literalInBody(?a):-"
        ----^
``` 
  

However when the atom `?a` additionally ocurrs in aother relation there is no error:
```
foo("test").
literalInBody(?a):-
    ?a = "test",
    foo(?a).
```

**Comment by mmarx** [comment]
We should consider equalities as positive atoms for the sake of checking safety. Note that this does not extend to inequalities, as, e.g., `p(?x) :- ?x != 23` is not safe.

**Comment by monsterkrampe** [comment]
This will be released in `0.4.0` now, right?
